system/services/presideObjects/PresideObjectService.cfc

```diff
@@ -459,11 +461,74 @@ component displayName="Preside Object Service" {
 			}
 
 			sqlAndParams.sql = ReReplaceNoCase( sqlAndParams.sql, ":#key#(\b)", ":#arguments.prefix##key#\1", "all" );
+
+			for ( var obfuscation in obfuscations ) {
+				_applyPrefixToObfuscatedSql( obfuscation=obfuscation, prefix=arguments.prefix, key=key );
+			}
+		}
+
+		for ( var obfuscation in obfuscations ) {
+			_applyNewObfuscatedSql( obfuscation );
+
+			sqlAndParams.sql = ReReplaceNoCase(
+				  sqlAndParams.sql
+				, obfuscation.encoded
+				, $obfuscateSqlForPreside( obfuscation.decoded )
+				, "all"
+			);
 		}
 
 		return sqlAndParams;
 	}
 
+	private array function _getObfuscationsInSql( 
+		  required string sql
+	) {
+		var obfuscations = [];
+		var obfsPattern  = "{{base64:([A-Za-z0-9\+\/=]+)}}";
+		var matches      = ReFindNoCase( obfsPattern, arguments.sql, 1, true, "all" );
+
+		for ( var matched in matches ) {
+			if ( !Len( matched.match[1] ) ) {
+				continue;
+			}
+			
+			var decoded = ToString( ToBinary( ReReplace( matched.match[1], obfsPattern, "\1" ) ) );
+			ArrayAppend( obfuscations, {
+				  encoded      = matched.match[1]
+				, decoded      = decoded
+				, obfuscations = _getObfuscationsInSql( decoded, arguments.cnt+1 )
```

Comment:
can't see `arguments.cnt` being defined anwhere. Also, let's not use `cnt` - `recursionCount` is clearer + doesn't look rude :p